### PR TITLE
W2 - nhatnguyen - Create screen with recyclerview

### PR DESCRIPTION
**Pull request by at-nhatnguyen**
#### Related issues (Optional)
Task  2

#### Description (Required)
- [x] create screen with recyclerview
- [x] use cardview

### CODE REVIEW

**Comment by at-nhatnguyen** [comment]
@at-huongnguyen please review my task, thank you

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    private void initViews() {
+        mRecycleViewComment = findViewById(R.id.recycleViewComment);
+    }
+
+    private void initData() {
+        mCommentList = new ArrayList<>();
+        mCommentList.add(new Comment("Nguyen Van A", "very good very good very good very good very good very good very good very good very good very good very good very good very good very good very good very good", 5));
+        mCommentList.add(new Comment("Nguyen Van B", "not good not good not good not good not good not good not good not good not good not good not good ", -4));
+        mCommentList.add(new Comment("Nguyen Van C", "not good", 3));
+        mCommentList.add(new Comment("Nguyen Van D", "very good", -2));
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "very good" appears 4 times in this file; the first occurrence is on line 36
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Comment by at-nhatnguyen** [comment]
@at-huongnguyen please review my task, thank you

File: app/src/at-nhatnguyen/AndroidManifest.xml [hunk 2]
```diff
@@ -9,13 +9,13 @@
...
         android:theme="@style/AppTheme">
-        <activity android:name="vn.asiantech.internship.MainActivity">
+        <activity android:name=".MainActivity">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".RecyclerviewActivity"></activity>
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/AndroidManifest.xml]: Update to: android:name=".RecyclerviewActivity"/>

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+import java.util.List;
+import vn.asiantech.internship.models.Comment;
+
+public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.CommentViewHolder> {
+    private List<Comment> mCommentLists;
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java]: Should rename `mCommentLists` variable to `mComments`

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    private OnItemClickListener mOnItemClickListener;
+
+    CommentAdapter(List<Comment> commentLists, OnItemClickListener onItemClickListener) {
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java]: Should rename `commentLists ` variable to `comments`

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    public CommentAdapter.CommentViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_comment, parent, false);
+        return new CommentViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(CommentViewHolder holder, int position) {
+        Comment comment = mCommentLists.get(position);
+        holder.mTvName.setText(comment.getName());
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java]: Create `onBindData` method and call this

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return mCommentLists.size();
+    }
+
+    public interface OnItemClickListener {
+        void onClickLike(int position);
+        void onClickDislike(int position);
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentAdapter.java]: Format code

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.models.Comment;
+
+public class RecyclerviewActivity extends AppCompatActivity implements CommentAdapter.OnItemClickListener {
+    private RecyclerView mRecycleViewComment;
+    private List<Comment> mCommentList;
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java]: Should rename mCommentLists variable to mComments

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    private CommentAdapter mCommentAdapter;
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java]: Should rename mCommentAdapter variable to mAdapter

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.models.Comment;
+
+public class RecyclerviewActivity extends AppCompatActivity implements CommentAdapter.OnItemClickListener {
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java]: Should rename RecyclerviewActivity class name to CommentActivity

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    private void initAdapter() {
+        mCommentAdapter = new CommentAdapter(mCommentList, this);
+        mRecycleViewComment.setLayoutManager(new LinearLayoutManager(this));
+        mRecycleViewComment.setAdapter(mCommentAdapter);
+    }
+
+    @Override
+    public void onClickLike(int position) {
+        mCommentList.get(position).clickLike();
+        mCommentAdapter.notifyDataSetChanged();
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java]: Use notifyItemChanged

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Override
+    public void onClickDislike(int position) {
+        mCommentList.get(position).clickDislike();
+        mCommentAdapter.notifyDataSetChanged();
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/RecyclerviewActivity.java]: Use notifyItemChanged

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/models/Comment.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
+package vn.asiantech.internship.models;
+
+public class Comment {
+    private String name;
+    private String commentContent;
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/models/Comment.java]: Rename `commentContent ` to `content`

File: app/src/at-nhatnguyen/res/layout/activity_recyclerview.xml [hunk 13]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.v7.widget.RecyclerView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:id="@+id/recycleViewComment"
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/res/layout/activity_recyclerview.xml]: Format code

File: app/src/at-nhatnguyen/res/layout/item_comment.xml [hunk 14]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-huongnguyen) [inline app/src/at-nhatnguyen/res/layout/item_comment.xml]: Remove ViewGroup

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentActivity.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+    private void initViews() {
+        mRecycleViewComment = findViewById(R.id.recycleViewComment);
+    }
+
+    private void initData() {
+        mComments = new ArrayList<>();
+        mComments.add(new Comment("Nguyen Van A", "very good very good very good very good very good very good very good very good very good very good very good very good very good very good very good very good", 5));
+        mComments.add(new Comment("Nguyen Van B", "not good not good not good not good not good not good not good not good not good not good not good ", -4));
+        mComments.add(new Comment("Nguyen Van C", "not good", 3));
+        mComments.add(new Comment("Nguyen Van D", "very good", -2));
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/CommentActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "very good" appears 4 times in this file; the first occurrence is on line 36
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Comment by at-nhatnguyen** [comment]
@at-huongnguyen please review my task, thank you

**Pull request merged**